studio/hooks/analytics/useLogsPreview.tsx

```diff
@@ -64,103 +63,99 @@ function useLogsPreview(
     refresh()
   }, [JSON.stringify(filters)])
 
-  // handle url generation for log pagination
-  const getKeyLogs: SWRInfiniteKeyLoader = (_pageIndex: number, prevPageData: Logs) => {
-    let queryParams
+  const queryParamsKey = genQueryParams(params as any)
 
-    // cancel request if no sql provided
-    if (!params.sql) {
-      // return null to restrict unnecessary requests to api
-      // https://swr.vercel.app/docs/conditional-fetching#conditional
-      return null
-    }
-
-    // if prev page data is 100 items, could possibly have more records that are not yet fetched within this interval
-    if (prevPageData === null) {
-      // reduce interval window limit by using the timestamp of the last log
-      queryParams = genQueryParams(params as any)
-    } else if ((prevPageData.result ?? []).length === 0) {
-      // no rows returned, indicates that no more data to retrieve and append.
-      return null
-    } else {
-      const len = prevPageData.result.length
-      const { timestamp: tsLimit }: LogData = prevPageData.result[len - 1]
-      const isoTsLimit = dayjs.utc(Number(tsLimit / 1000)).toISOString()
-      // create new key from params
-      queryParams = genQueryParams({ ...params, iso_timestamp_end: isoTsLimit } as any)
+  const {
+    data,
+    isLoading,
+    isRefetching,
+    error: rqError,
+    fetchNextPage,
+    isFetchingNextPage,
+    refetch,
+  } = useInfiniteQuery(
+    ['projects', projectRef, 'logs', queryParamsKey],
+    ({ signal, pageParam }) => {
+      return get<Logs>(
+        `${API_URL}/projects/${projectRef}/analytics/endpoints/logs.all?${genQueryParams({
+          ...params,
+          iso_timestamp_end: pageParam,
```

Comment:
pageParam doesn't factor in initial hook parameters

see here https://tanstack.com/query/v4/docs/react/guides/infinite-queries#what-if-i-need-to-pass-custom-information-to-my-query-function

Reply:
When loading from query parameters, the query string should be parsed for `its` and `ite` in the parent component and passed as the initial `params` arg to the hook, set as `filters` in the local state, and sent in the initial request.

What is happening is that the local filters that cache the subsequent updates to `params` are getting ignored when building requests, because you are using the `...params`, hence it always uses the initial filters.


The "Loading older" logic is correct.